src/core/interp.c

```diff
@@ -3355,7 +3355,7 @@ void MVM_interp_run(MVMThreadContext *tc, void (*initial_invoke)(MVMThreadContex
                 if (REPR(maybe_cu)->ID == MVM_REPR_ID_MVMCompUnit) {
                     MVMCompUnit *cu = (MVMCompUnit *)maybe_cu;
                     if (cu->body.mainline_frame) {
-                        MVMObject *coderef;
+                        MVMObject *coderef = 0;
```

Comment:
But....wow, this is such an obvious bug it's kinda embarrassing that I didn't see it. I even had a look at this warning and it confused me endlessly, because there's just no way a longjmp or vfork could happen between initialization and use of the value. But actually it can, because initialization happens conditionally within the following loop. So the variable may be initialized in a compunitmainline op, then a longjmp clobbers it, then we run compunitmainline again and this time don't find a mainline_frame. It's a purely theoretical one, but a bug nonetheless. Thanks for finally solving this mystery!

One nit to pick though: while the = 0 is technically correct, we usually spell out NULL in MoarVM and I find that just more explicit.

Reply:
I'll change to NULL then.